manifests/repo.pp

```diff
@@ -93,7 +88,10 @@
           }
         } else {
           $operatingsystem = downcase($::operatingsystem)
-
+          case $::operatingsystemrelease {
+            /\/sid$/ : { $releasename = regsubst($::operatingsystemrelease, '/sid$', '') }
+            default  : { $releasename = $::lsbdistcodename }
```

Comment:
Ionly know that it requires the lsb-release package installed and in general the facts are only present on Debian based systems. I was looking for alternative core facts but didn't find any :(